app/src/main/java/helium314/keyboard/latin/ClipboardHistoryManager.kt

```diff
@@ -21,9 +21,10 @@ class ClipboardHistoryManager(
 
     fun onCreate() {
         clipboardManager = latinIME.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
-        fetchPrimaryClip()
         clipboardManager.addPrimaryClipChangedListener(this)
-        loadPinnedClips()
+        if (historyEntries.isEmpty())
+            loadPinnedClips()
+        onPrimaryClipChanged()
```

Comment:
But why should we fetch the primary clipboard in OnCreate() and add it to histroyEntries if the user disabled the internal clipboard history?
you could replace the line in question with the current content of onPrimaryClipChanged() but I don't see how changing that function would break things because it may be called at any time by the listener

Reply:
> you could replace the line in question with the current content of onPrimaryClipChanged() but I don't see how changing that function would break things because it may be called at any time by the listener

By a listener that listens to clipboard changes.
If I decide to show a message "copied <content> to clipboard" when the primary clip is changed (and such thing is wanted by some user iirc), this message should _not_ be shown onCreate.

Reply:
Oh if that's the case it may make sense.
You should know though that in recent OneUI versions there is a built in toast for this so there will be a conflict if Heliboard also sends a notification.
Such change would also require adding the notification permission, which is not really a problematic one, but I think we can do without it.
In the #679 I added a subtle animation to the copy (and the proposed clear clipboard history) toolbar keys so the user has some visual feedback that the key has been pressed if his android flavor does not have have an indication for the clipboard change.

Reply:
So yes, please simply copy the content of `onPrimaryClipChanged` instead of calling the function.

Alternatively you could maybe move the `mClipboardHistoryEnabled` check into `fetchPrimaryClip`. At least on first glance it would make sense to me.

> Such change would also require adding the notification permission

It's also possible to show some sort of fake toast message in a normal (usually hidden) view over the keyboard. (I have a todo to try it, but way too much work and this is not worth the effort for potentially larger changes)
I agree we can do without the notification permission; I want to avoid additional permissions wherever possible.

Reply:
Ok i've replaced the code in question 
I'd personally not like a fake toast because my device already shows a real toast
Btw there is another problem with clipboard, it coerces clipboard images to text which can cause a freeze when opening the clipboard view.
I've applied a fix in #647 (clipboard suggestions) to prevent the keyboard from becoming unresponsive when an image is copied to the clipboard